src/cryptonote_core/tx_pool.cpp

```diff
@@ -1416,6 +1422,21 @@ namespace cryptonote
     return ret;
   }
   //---------------------------------------------------------------------------------
+  bool tx_memory_pool::is_transaction_meta_ready_to_go(const txpool_tx_meta_t& txd) const
+  {
+    // For txs (FCMP++ txs at time of writing) where it is guaranteed that they will fail
+    // verification if a certain height in the chain is not both present and confirmed, we can abort
+    // early, skipping a blob load, a deserialization, and a call to check_tx_inputs().
+    if (txd.hard_height_requirement)
+    {
+      const uint64_t curr_height = m_blockchain.get_current_blockchain_height();
+      if (txd.max_used_block_height + CRYPTONOTE_DEFAULT_TX_SPENDABLE_AGE > curr_height)
```

Comment:
Idea: if you have `max_used_block_height=reference_block` for FCMP++ txs, you can just check `txd.max_used_block_height >= curr_height` here

We wouldn't need the `hard_height_requirement` field, and can call this for any tx. It also avoids any issues surrounding off-by-1 logic

It introduces a difference on what `max_used_block_height` means for pre and post FCMP++ txs (which affects the if for `// enforce min output age` in `blockchain.cpp`), so I would accept if you prefer not to do that

Reply:
Good point, you're right. As is is ok with me, will review closer for off-by-1

Reply:
Looks good, always tricky